raijin_scripts/execute_notify_on_error_ds/run

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -eu
+
+while [[ $# -gt 0 ]]; do
+    key="$1"
+    case ${key} in
+        --dea-module )          shift
+                                MODULE=$1
+                                ;;
+        --awsprofile )          shift
+                                AWS_PROFILE=$1
+                                ;;
+        --recemail )            shift
+                                RECP=$1
+                                ;;
+        * )                     exit 0
+    esac
+    shift
+done
+
+module use /g/data/v10/public/modules/modulefiles
+module load "$MODULE"
+
+# If aws cli is not installed, exit
+aws --version 2> /dev/null || exit 0;
+
+# Get the latest csv file from the coherence directory
+file=$(find /g/data/v10/work/coherence/ -type f -name '*.csv' -print | sort | tail -n -1)
+num_rows=$(head "$file" | wc -l)
+
+# Check if we have any bad datasets in the csv file
+[ "$num_rows" -lt 2 ] && echo "No bad datasets found within the database" && exit 0;
+
+filename=~/.aws/credentials
+
+datetime=$(date '+%F-%T')
+
+# -s: Returns true if file exists and has a size > 0
+if [[ -s ~/.aws/credentials ]]; then
+    exists=
+    profilename="$AWS_PROFILE"
+    while read -ra line; 
+    do
+        for word in "${line[@]}";
+        do
+            [ "$word" = "[$profilename]" ] && exists="$word"
+        done;
+    done < "$filename"
+
+    if [ "$exists" ]
+    then
+        aws configure list --profile "$profilename"
+
+        s3_file=s3://dea-lambda/serverless/coherence/error_datasets_$(date '+%F').csv
+
+        # Upload to aws s3
+        aws s3 --profile "$profilename" cp "$file" "$s3_file"
+
+        # Email Content
+        email_content="$datetime
+
+         Sub: Erroneous datasets found in database (Reported by coherence tool)
+
+         NCI file path: ${file}
+         AWS S3 url   : s3://dea-lambda/serverless/coherence/error_datasets_$(date '+%F').csv
+
+         "
+
+        # Send Email to the recepient email ID (mailx will not work in Raijin Login nodes)
+        echo "$email_content" | mailx -s "Erroneous datasets detected ($datetime)" "$RECP"
+        MSG="$email_content"
+
+        PAYLOAD="payload={\"text\": \"$MSG\"}"
+        DEA_ORCHESTRATION_WEBHOOK=https://hooks.slack.com/services/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
+
+        # Send slack notification to '#dea-orchestration' channel about new CSV file that has been pushed to s3 bucket
+        curl -X POST --data-urlencode "$PAYLOAD" "$DEA_ORCHESTRATION_WEBHOOK"
+    else
+        echo "'$profilename' profile does not exist in '~/.aws/credentials' file"
+        exit 0
```

Comment:
We want this to have a *non* zero exit code.

It should get passed back through lambda and cloud watch so that we get a notification running this script has failed.

Reply:
Implemented in `PR#76`